api/controllers/MerchStoreController.ts

```diff
@@ -158,14 +159,27 @@ export class MerchStoreController {
 
   @UseBefore(UserAuthentication)
   @Post('/item/picture/:uuid')
-  async updateMerchPhoto(@UploadedFile('image',
+  async createMerchItemPhoto(@UploadedFile('image',
     { options: StorageService.getFileOptions(MediaType.MERCH_PHOTO) }) file: File,
     @Params() params: UuidParam,
-    @AuthenticatedUser() user: UserModel): Promise<UpdateMerchPhotoResponse> {
+    @AuthenticatedUser() user: UserModel): Promise<CreateMerchPhotoResponse> {
     if (!PermissionsService.canEditMerchStore(user)) throw new ForbiddenError();
-    const picture = await this.storageService.upload(file, MediaType.MERCH_PHOTO, params.uuid);
-    const item = await this.merchStoreService.editItem(params.uuid, { picture });
-    return { error: null, item };
+
+    // generate a random string for the picture url
+    const randomID = StorageService.getRandomString();
+    const picture = await this.storageService.uploadToFolder(file, MediaType.MERCH_PHOTO, randomID, params.uuid);
+    const photo = await this.merchStoreService.createItemPhoto(params.uuid, { picture });
+
+    return { error: null, photo };
+  }
+
+  @UseBefore(UserAuthentication)
+  @Delete('/item/picture/:uuid')
+  async deleteMerchItemPhoto(@Params() params: UuidParam, @AuthenticatedUser() user: UserModel):
+  Promise<DeleteMerchItemPhotoResponse> {
+    if (!PermissionsService.canEditMerchStore(user)) throw new ForbiddenError();
+    await this.merchStoreService.deleteItemPhoto(params.uuid);
+    return { error: null };
   }
```

Comment:
Shouldn't we also delete the file from S3?

Reply:
wait a sec, I don't think the photo was deleted from the original code base

Reply:
Not sure what you mean here, can you elaborate?

Reply:
there wasn't code to delete photo from S3 before I work on this branch, oddly